src/test/regress/greenplum_schedule

```diff
@@ -258,7 +258,7 @@ test: ao_locks
 test: freeze_aux_tables
 
 # cbdb parallel test
-test: cbdb_parallel
+ignore: cbdb_parallel
```

Comment:
Why we ignore cbdb_parallel here?

Reply:
From the comments, disable due to unstable results. LGTM

Reply:
Hmm..why previous CI doesn't have this problem? Could you show the detailed failed case info.

Reply:
@avamingli - The previous setup utilized GitHub self-hosted runners running on AWS-hosted VMs. Currently, we are working on running individual schedules in parallel, each within its own dedicated container (with limited CPU, sysctl configuration, and disk resources), executed on GitHub-hosted runners.

Despite this setup, as well as previous container and VM-based test executions I’ve observed, the test execution rates remain somewhat inconsistent. We do have basic support in place to start monitoring and identifying which tests are exhibiting inconsistent behavior.

I am actively working on re-introducing the test suites (cbdb_parallel and instr_in_shmem_verify), which I temporarily commented out. I will attempt to reintroduce them in some form.

Reply:
@avamingli I have invited you to collaborate with me on my cloudbery project. This will allow you to review my test run where a github action reproduced the issue. The `cbdb_parallel` test is failing in the containers due to "No space left on device". Is this a result of running out of disk space or memory?

You should be able to review the test logs here:

https://github.com/edespino/cloudberry/actions/runs/11924771340

```
diff -I HINT: -I CONTEXT: -I GP_IGNORE: -U3 /__w/cloudberry/cloudberry/src/test/regress/expected/cbdb_parallel.out /__w/cloudberry/cloudberry/src/test/regress/results/cbdb_parallel.out
--- /__w/cloudberry/cloudberry/src/test/regress/expected/cbdb_parallel.out	2024-11-19 17:23:58.142198133 -0800
+++ /__w/cloudberry/cloudberry/src/test/regress/results/cbdb_parallel.out	2024-11-19 17:23:58.206197545 -0800
@@ -536,11 +536,8 @@
 GP_IGNORE:(20 rows)
 
 select count(*) from ao1, ao2 where ao1.x = ao2.x;
-  count  
----------
- 1080000
-(1 row)
-
+ERROR:  could not resize shared memory segment "/PostgreSQL.2418892932" to 8388608 bytes: No space left on device
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
 commit;
 --
 -- test parallel with indices
@@ -1563,45 +1560,13 @@
 GP_IGNORE:(30 rows)
 
 select * from rt1  join (select count(*) as c, sum(t1.a) as a  from t1 join t2 using(a)) t3 on t3.c = rt1.a;
- a | b | c | a 
----+---+---+---
-(0 rows)
-
+ERROR:  could not resize shared memory segment "/PostgreSQL.4177304382" to 4194304 bytes: No space left on device
 set local enable_parallel = off;
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
 explain(costs off, locus) select * from rt1  join (select count(*) as c, sum(t1.a) as a  from t1 join t2 using(a)) t3 on t3.c = rt1.a;
-QUERY PLAN
-___________
- Hash Join
-   Locus: Entry
-   Hash Cond: (rt1.a = (count(*)))
-   ->  Gather Motion 1:1  (slice1; segments: 1)
-         Locus: Entry
-         ->  Seq Scan on rt1
-               Locus: SegmentGeneral
-   ->  Hash
-         Locus: Entry
-         ->  Finalize Aggregate
-               Locus: Entry
-               ->  Gather Motion 3:1  (slice2; segments: 3)
-                     Locus: Entry
-                     ->  Partial Aggregate
-                           Locus: Hashed
-                           ->  Hash Join
-                                 Locus: Hashed
-                                 Hash Cond: (t1.a = t2.a)
-                                 ->  Seq Scan on t1
-                                       Locus: Hashed
-                                 ->  Hash
-                                       Locus: Hashed
-                                       ->  Seq Scan on t2
-                                             Locus: Hashed
-GP_IGNORE:(25 rows)
-
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
 select * from rt1  join (select count(*) as c, sum(t1.a) as a  from t1 join t2 using(a)) t3 on t3.c = rt1.a;
- a | b | c | a 
----+---+---+---
-(0 rows)
-
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
 abort;
 --
 -- Test final join path's parallel_workers should be same with join_locus whose
@@ -2247,44 +2212,24 @@
 GP_IGNORE:(10 rows)
 
 select sum(t1.c1) from t1 where c1 not in (select c2 from t2);
- sum 
------
-   1
-(1 row)
-
+ERROR:  could not resize shared memory segment "/PostgreSQL.3791510102" to 8388608 bytes: No space left on device
 explain(costs off) select * from t1 where c1 not in (select c2 from t3_null);
-QUERY PLAN
-___________
- Gather Motion 6:1  (slice1; segments: 6)
-   ->  Parallel Hash Left Anti Semi (Not-In) Join
-         Hash Cond: (t1.c1 = t3_null.c2)
-         ->  Parallel Seq Scan on t1
-         ->  Parallel Hash
-               ->  Broadcast Workers Motion 6:6  (slice2; segments: 6)
-                     ->  Parallel Seq Scan on t3_null
-GP_IGNORE:(8 rows)
-
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
 select * from t1 where c1 not in (select c2 from t3_null);
- c1 | c2 
-----+----
-(0 rows)
-
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
 -- non-parallel results.
 set local enable_parallel = off;
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
 select sum(t1.c1) from t1 where c1 not in (select c2 from t2);
- sum 
------
-   1
-(1 row)
-
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
 select * from t1 where c1 not in (select c2 from t3_null);
- c1 | c2 
-----+----
-(0 rows)
-
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
 drop table t1;
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
 drop table t2;
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
 drop table t3_null;
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
 end;
 --
 -- End of Test Parallel-aware Hash Left Anti Semi (Not-In) Join.
@@ -2551,19 +2496,15 @@
 begin;
 set local max_parallel_workers_per_gather = 8;
 select * from refresh_compare(true, false);
- parallel_is_better 
---------------------
- t
-(1 row)
-
+ERROR:  could not resize shared memory segment "/PostgreSQL.2096820432" to 8388608 bytes: No space left on device
+CONTEXT:  SQL statement "refresh materialized view matv"
+PL/pgSQL function refresh_compare(boolean,boolean) line 30 at SQL statement
 select * from refresh_compare(false, false);
- parallel_is_better 
---------------------
- t
-(1 row)
-
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
 drop function refresh_compare;
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
 reset max_parallel_workers_per_gather;
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
 end;
 --
 -- Parallel Create AO/AOCO Table AS

```

Reply:
> The cbdb_parallel test is failing in the containers due to "No space left on device". Is this a result of running out of disk space or memory?

I think so, we never have such error before.

Reply:
when I make gpdb test run in GitHub Actions I met this before.
my memo
```
GitHub Actions has a lot of limitations, and is basically unavailable to businesses that want to use it without paying for it.
Recently, I tried to run gpdb on GitHub Actions and encountered a lot of difficulties:
1. there is not enough space, free space is only about 10G or so, the solution is to delete some services, for example, Android and .net take up about 20G by default, llvm can also be deleted. 2. memory is not enough, without paying for it, it is basically impossible for enterprises to use it.
2. memory is not enough, if you don't pay for it, the available memory is 7.3G, running a gpdb service is about 2G, and some parallel tests will hang by default.
3. because of the above hardware problems, time is not enough to run a complete set of tests 2 hours is difficult to run, but also some tests because of environmental problems will hang
4. hijack will be difficult and security issues, debug more troublesome
---
The advantages are
1. trigger stability, friendly to open source users
2. compilation is quite fast
3. can be given to some extensions with the need for some trick twice actions/checkout

```
the solution is to get more disk space:
you can use  Actions like https://github.com/marketplace/actions/maximize-build-disk-space
or buy a better Actions on GitHub

Reply:
I downloaded the regression logs from the github action and it seems we ran out of shared memory:

`2024-11-19 17:22:37.539475 PST,"gpadmin","regression",p10668,th-111919872,"[local]",,2024-11-19 17:22:33 PST,819,con36,cmd142,seg-1,,dx15,x819,sx1,"ERROR","53100","could not resize shared memory segment ""/PostgreSQL.2418892932"" to 8388608 bytes: No space left on device  (seg0 slice1 172.18.0.2:7002 pid=10670)",,,,,,"select count(*) from ao1, ao2 where ao1.x = ao2.x;",0,,"dsm_impl.c",317,
`

Reply:
@yihong0618 (GOOD BUDDY!) Thank you for sharing these detailed observations. Eventually, we might turn to the Apache Software Foundation (ASF) Infrastructure (INFRA) team to allocate additional resources. As we start this Apache journey, I don't want to jump to the decision to throw more resources without ensuring we absolutely can't work within the initial confines. We are not a business, we are an open source community. We need to keep our community productive and hopefully we can strike a good balance.

Regarding disk space, I've seen similar guidance on freeing up space and I believe we have sufficient disk space for the basic installcheck schedules. It's the shared memory limitations that we are currently encountering.